pkg/core/interop/runtime/witness.go

```diff
@@ -69,9 +69,7 @@ func checkScope(ic *interop.Context, hash util.Uint160) (bool, error) {
 				return true, nil
 			}
 			if c.Scopes&transaction.CalledByEntry != 0 {
-				callingScriptHash := ic.VM.GetCallingScriptHash()
-				entryScriptHash := ic.VM.GetEntryScriptHash()
-				if callingScriptHash.Equals(util.Uint160{}) || callingScriptHash == entryScriptHash {
+				if ic.VM.IstackLen() <= 2 { // Entry -> 1, called contract -> 2.
```

Comment:
This is wrong.

Reply:
Could you, please, clarify, why calling/entry script hash equality check is removed from the C# implementation?

Reply:
https://github.com/neo-project/neo/pull/2756#issuecomment-1188844828
This doesn't do anything useful without neo-project/neo#2756 though, so it might as well be omitted.